pkg/agent/baker.go

```diff
@@ -192,6 +218,33 @@ func (t *TemplateGenerator) getBakerFuncMap(config *NodeBootstrappingConfigurati
 	return funcMap
 }
 
+// normalizeResourceGroupNameForLabel normalizes resource group name to be used as a label,
+// similar to what the ARM template used to do.
+//
+// When ARM template was used, the following is used:
+//   variables('labelResourceGroup')
+// which is defined as:
+//   [if(or(or(endsWith(variables('truncatedResourceGroup'), '-'), endsWith(variables('truncatedResourceGroup'), '_')), endsWith(variables('truncatedResourceGroup'), '.')), concat(take(variables('truncatedResourceGroup'), 62), 'z'), variables('truncatedResourceGroup'))]
+// the "truncatedResourceGroup" is defined as:
+//   [take(replace(replace(resourceGroup().name, '(', '-'), ')', '-'), 63)]
+// This function does the same processing.
+func normalizeResourceGroupNameForLabel(resourceGroupName string) string {
+	truncated := resourceGroupName
+	truncated = strings.ReplaceAll(truncated, "(", "-")
+	truncated = strings.ReplaceAll(truncated, ")", "-")
+	const maxLen = 63
+	if len(truncated) > maxLen {
+		truncated = truncated[0:maxLen]
+	}
+
+	if strings.HasSuffix(truncated, "-") ||
+		strings.HasSuffix(truncated, "_") ||
+		strings.HasSuffix(truncated, ".") {
+		return truncated[0:len(truncated)-1] + "z"
```

Comment:
more precisely:

if len(truncated) > 62 {
 return truncated[0:len(truncated)-1] + "z"
} else {
return truncated + "z"
}
}

Reply:
This is to match the ARM template that sets labelResourceGroup to:
[if(or(or(endsWith(variables('truncatedResourceGroup'), '-'), endsWith(variables('truncatedResourceGroup'), '_')), endsWith(variables('truncatedResourceGroup'), '.')), concat(take(variables('truncatedResourceGroup'), 62), 'z'), variables('truncatedResourceGroup'))]

What it does is: it only replaces the last character with z if it is '-', '_', or '.'.

If truncatedResourceGroup does not end with any of those 3, then
    labelResourceGroup = truncatedResourceGroup

Reply:
you're missing the conversion of take(variables('truncatedResourceGroup'), 62), where it get max of 62 char  so no trimming for short strings

e.g if the string is "abc-" and the final string will be "abc-z" ,

Reply:
Fixed now.